src/Submission/ReduxProjectSubmission.js

```diff
@@ -36,24 +38,57 @@ const submitToServer = (fullProject, methodIn = 'PUT') => (dispatch, getState) =
     // remove line break in json file
     file = file.replace(/\n/g, '');
   }
+
+  let i = 0;
+  if (submission.file_type === 'text/tab-separated-values') {
+    const fileSplited = file.split(/\n/g);
+    let fileHeader = fileSplited.shift();
+    fileHeader += '\r';
+    if (fileSplited.length > lineLimit) {
+      i = lineLimit;
+      let fileChunk = fileHeader;
+      while (fileSplited.length > 0) {
+        fileChunk += fileSplited.shift();
```

Comment:
I think using `.shift()` might not be the most efficient way to iterate through the array:

> To remove the returned item without re-addressing the array and invalidating all references to it, shift() requires moving the entire array around; (https://stackoverflow.com/questions/6501160/why-is-pop-faster-than-shift)

Do you need to delete entries from `fileSplited`, or can you just iterate through the array without modifying it?

Reply:
Right! This has been resolved by only iterating through `fileSplited` array